packages/valory/skills/proposal_collector_abci/behaviours.py

```diff
@@ -240,93 +250,69 @@ def _get_active_proposals(self) -> Generator[None, None, str]:
-            )
-
-            # Select the first proposal
-            proposal_id = sorted_proposals[0]["id"] if sorted_proposals else None
-
-            # Check whether we have delegations
-            if not proposal_id:
-                proposal_id = SelectProposalRound.NO_PROPOSAL
+        active_proposals = filter(
+            lambda ap: ap["end"]["number"] > current_block, active_proposals
+        )
 
-            sender = self.context.agent_address
-            payload = SelectProposalPayload(sender=sender, proposal_id=proposal_id)
+        # Update the current proposals
+        proposals = self.synchronized_data.proposals
+
+        # Set all the current proposals "votable" field to False
+        for proposal in proposals.values():
+            proposal["votable"] = False
+
+        for active_proposal in active_proposals:
+            if active_proposal["id"] not in proposals:  # This is a new proposal
+                proposals[active_proposal["id"]] = {
+                    **active_proposal,
+                    "votable": True,
+                    "vote_intention": None,
+                    "vote": None,  # We have not voted for this one yet
+                    "remaining_blocks": active_proposal["end"]["number"]
+                    - current_block,
+                }
+            else:
+                # The proposal is still votable
+                proposals[active_proposal["id"]]["votable"] = True
+                proposals[active_proposal["id"]]["remaining_blocks"] = (
+                    proposals[active_proposal["id"]]["end"]["number"] - current_block
+                )
 
-            self.context.logger.info(f"proposal_id = {proposal_id}")
+        # Get all the governors from the current delegations
+        delegation_governors = [
+            d["governor_address"] for d in self.synchronized_data.delegations
+        ]
+
+        # Get the proposals where we can vote:
+        # - Votable
+        # - Not voted before
+        # - Governor in the delegation list
+        votable_proposals = (
+            proposal
+            for proposal in proposals.values()
+            if proposal["votable"]
+            and not proposal["vote"]
+            and proposal["governor"]["id"].split(":")[-1] in delegation_governors
+        )
 
-        with self.context.benchmark_tool.measure(self.behaviour_id).consensus():
-            yield from self.send_a2a_transaction(payload)
-            yield from self.wait_until_round_end()
+        # Sort votable_proposals by vote end block number, in ascending order
+        votable_proposal_ids = [
+            p["id"]
+            for p in sorted(votable_proposals, key=lambda p: p["remaining_blocks"])
+        ]
 
-        self.set_done()
+        return json.dumps(
+            {
+                "proposals": proposals,
+                "votable_proposal_ids": votable_proposal_ids,
+            },
+            sort_keys=True,
+        )
```

Comment:
Why keep `proposals` and `votable_proposal_ids` and not `votable_proposals` only? This would also resolve the previous comment (https://github.com/valory-xyz/governatooorr/pull/22/files#r1161612875).

Reply:
Addressed on https://github.com/valory-xyz/governatooorr/pull/22#discussion_r1161618890.